sound/soc/sof/hw-spi.c

```diff
@@ -158,7 +158,7 @@ static irqreturn_t spi_irq_handler(int irq __maybe_unused, void *context)
 	// on SPI based devices this will likely come via a SoC GPIO IRQ
 
 	// check if GPIO is assetred and if so run thread.
-	if (sof_pdata->gpio >= 0 &&
+	if (sof_pdata->gpio > 0 &&
```

Comment:
sof_pdata->gpio = desc_to_gpio(gpiod)

int desc_to_gpio(const struct gpio_desc *desc)
{
	return desc->gdev->base + (desc - &desc->gdev->descs[0]);
}

so gpio should be the type of in, not unsigned in

Reply:
@RanderWang but is it ever likely to return anything < 0?

Reply:
@RanderWang but is it ever likely to return anything < 0?

Reply:
@ranj063 @RanderWang How about using gpio_is_valid(sof_pdata->gpio)? And, of cause sof_pdata->gpio should be int.

Reply:
thanks for the comments, I just changed the type in sof.h, it's enough to get the warnings out of the way.